arangod/Cluster/SynchronizeShard.cpp

```diff
@@ -1020,12 +873,12 @@ bool SynchronizeShard::first() {
       auto lastTick =
         arangodb::basics::VelocyPackHelper::getNumericValue<TRI_voc_tick_t>(
           sy, LAST_LOG_TICK, 0);
-      VPackBuilder builder;
 
       ResultT<TRI_voc_tick_t> tickResult =
-        catchupWithReadLock(ep, database, *collection, clientId, shard,
-                            leader, lastTick, builder);
+          catchupWithReadLock(ep, database, *collection, clientId, shard,
+                              leader, lastTick, tailingSyncer);
       if (!tickResult.ok()) {
+        tailingSyncer->unregisterFromLeader();
```

Comment:
Perhaps this should be done using a scopeGuard since there are two other places where we call `unregisterFromLeader`?

Reply:
I agree this would work too and will look "nicer". But IMHO it is a "nice-to-have" change which will not affect the behavior.
So I suggest doing all the cleanup work you suggested just for devel, in a separate PR.